php grpc ext fails to compile (php 7.0.8, ubuntu 16.04)

>  `$
> 
>  make       
> /bin/bash /var/www/grpc/src/php/ext/grpc/libtool --mode=compile cc -Wall -Werror -std=c11 -I. -I/var/www/grpc/src/php/ext/grpc -DPHP_ATOM_INC -I/var/www/grpc/src/php/ext/grpc/include -I/var/www/grpc/src/php/ext/grpc/main -I/var/www/grpc/src/php/ext/grpc -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include  -DHAVE_CONFIG_H  -g -O2   -c /var/www/grpc/src/php/ext/grpc/call.c -o call.lo 
> libtool: compile:  cc -Wall -Werror -std=c11 -I. -I/var/www/grpc/src/php/ext/grpc -DPHP_ATOM_INC -I/var/www/grpc/src/php/ext/grpc/include -I/var/www/grpc/src/php/ext/grpc/main -I/var/www/grpc/src/php/ext/grpc -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -I/usr/local/include -DHAVE_CONFIG_H -g -O2 -c /var/www/grpc/src/php/ext/grpc/call.c  -fPIC -DPIC -o .libs/call.o
> /var/www/grpc/src/php/ext/grpc/call.c: In function ‘grpc_parse_metadata_array’:
> cc1: error: function may return address of local variable [-Werror=return-local-addr]
> In file included from /var/www/grpc/src/php/ext/grpc/php_grpc.h:60:0,
>                  from /var/www/grpc/src/php/ext/grpc/call.h:44,
>                  from /var/www/grpc/src/php/ext/grpc/call.c:34:
> /var/www/grpc/src/php/ext/grpc/php7_wrapper.h:146:8: note: declared here
>    zval _stack_zval_##pzv; \
>         ^
> /var/www/grpc/src/php/ext/grpc/call.c:89:3: note: in expansion of macro ‘PHP_GRPC_MAKE_STD_ZVAL’
>    PHP_GRPC_MAKE_STD_ZVAL(array);
>    ^
> cc1: error: function may return address of local variable [-Werror=return-local-addr]
> /var/www/grpc/src/php/ext/grpc/php7_wrapper.h:146:8: note: declared here
>    zval _stack_zval_##pzv; \
>         ^
> /var/www/grpc/src/php/ext/grpc/call.c:89:3: note: in expansion of macro ‘PHP_GRPC_MAKE_STD_ZVAL’
>    PHP_GRPC_MAKE_STD_ZVAL(array);
>    ^
> /var/www/grpc/src/php/ext/grpc/call.c: In function ‘grpc_php_wrap_call’:
> /var/www/grpc/src/php/ext/grpc/call.c:190:10: error: function returns address of local variable [-Werror=return-local-addr]
>    return call_object;
>           ^
> In file included from /var/www/grpc/src/php/ext/grpc/php_grpc.h:60:0,
>                  from /var/www/grpc/src/php/ext/grpc/call.h:44,
>                  from /var/www/grpc/src/php/ext/grpc/call.c:34:
> /var/www/grpc/src/php/ext/grpc/php7_wrapper.h:146:8: note: declared here
>    zval _stack_zval_##pzv; \
>         ^
> /var/www/grpc/src/php/ext/grpc/call.c:185:3: note: in expansion of macro ‘PHP_GRPC_MAKE_STD_ZVAL’
>    PHP_GRPC_MAKE_STD_ZVAL(call_object);
>    ^
> cc1: all warnings being treated as errors
> Makefile:196: recipe for target 'call.lo' failed
> make: *** [call.lo] Error 1
> 
> `

I tried `v1.0.x`, same result

--- a/examples/php/route_guide/README.md
+++ b/examples/php/route_guide/README.md
@@ -1,6 +1,6 @@
 #gRPC Basics: PHP sample code
 
 The files in this folder are the samples used in [gRPC Basics: PHP][],
-a detailed tutorial for using gRPC in Ruby.
+a detailed tutorial for using gRPC in PHP.
 
 [gRPC Basics: PHP]:http://www.grpc.io/docs/tutorials/basic/php.html

--- a/package.xml
+++ b/package.xml
@@ -10,7 +10,7 @@
   <email>grpc-packages@google.com</email>
   <active>yes</active>
  </lead>
- <date>2016-07-28</date>
+ <date>2016-08-09</date>
  <time>16:06:07</time>
  <version>
   <release>1.0.0</release>
@@ -22,7 +22,7 @@
  </stability>
  <license>BSD</license>
  <notes>
-- PHP7 Support continued, reduce code duplication #7543
+- Fixed Ubuntu compile error #7571, #7642
  </notes>
  <contents>
   <dir baseinstalldir="/" name="/">
@@ -1129,6 +1129,21 @@ Update to wrap gRPC C Core version 0.10.0
    <license>BSD</license>
    <notes>
 - PHP7 Support continued, reduce code duplication #7543
+   </notes>
+  </release>
+  <release>
+   <version>
+    <release>1.0.0RC4</release>
+    <api>1.0.0RC4</api>
+   </version>
+   <stability>
+    <release>stable</release>
+    <api>stable</api>
+   </stability>
+   <date>2016-08-09</date>
+   <license>BSD</license>
+   <notes>
+- Fixed Ubuntu compile error #7571, #7642
    </notes>
   </release>
  </changelog>

--- a/src/php/ext/grpc/php7_wrapper.h
+++ b/src/php/ext/grpc/php7_wrapper.h
@@ -143,8 +143,7 @@ static inline int php_grpc_zend_hash_find(HashTable *ht, char *key, int len,
 
 #define PHP_GRPC_RETURN_STRING(val, dup) RETURN_STRING(val)
 #define PHP_GRPC_MAKE_STD_ZVAL(pzv) \
-  static zval _stack_zval_##pzv; \
-  pzv = &(_stack_zval_##pzv)
+  pzv = (zval *)emalloc(sizeof(zval));
 #define PHP_GRPC_DELREF(zv)
 
 #define PHP_GRPC_WRAP_OBJECT_START(name) \

--- a/src/php/ext/grpc/php7_wrapper.h
+++ b/src/php/ext/grpc/php7_wrapper.h
@@ -143,7 +143,7 @@ static inline int php_grpc_zend_hash_find(HashTable *ht, char *key, int len,
 
 #define PHP_GRPC_RETURN_STRING(val, dup) RETURN_STRING(val)
 #define PHP_GRPC_MAKE_STD_ZVAL(pzv) \
-  zval _stack_zval_##pzv; \
+  static zval _stack_zval_##pzv; \
   pzv = &(_stack_zval_##pzv)
 #define PHP_GRPC_DELREF(zv)
 

--- a/templates/package.xml.template
+++ b/templates/package.xml.template
@@ -12,7 +12,7 @@
     <email>grpc-packages@google.com</email>
     <active>yes</active>
    </lead>
-   <date>2016-07-28</date>
+   <date>2016-08-09</date>
    <time>16:06:07</time>
    <version>
     <release>${settings.php_version.php()}</release>
@@ -24,7 +24,7 @@
    </stability>
    <license>BSD</license>
    <notes>
-  - PHP7 Support continued, reduce code duplication #7543
+  - Fixed Ubuntu compile error #7571, #7642
    </notes>
    <contents>
     <dir baseinstalldir="/" name="/">
@@ -247,6 +247,21 @@
      <license>BSD</license>
      <notes>
   - PHP7 Support continued, reduce code duplication #7543
+     </notes>
+    </release>
+    <release>
+     <version>
+      <release>1.0.0RC4</release>
+      <api>1.0.0RC4</api>
+     </version>
+     <stability>
+      <release>stable</release>
+      <api>stable</api>
+     </stability>
+     <date>2016-08-09</date>
+     <license>BSD</license>
+     <notes>
+  - Fixed Ubuntu compile error #7571, #7642
      </notes>
     </release>
    </changelog>

